lib/spack/spack/cmd/create.py

```diff
@@ -637,6 +645,31 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-b', '--batch', action='store_true',
         help="don't ask which versions to checksum")
+    subparser.add_argument(
+        '-g', '--git', action='store', nargs='?', default='#AUTO-GIT-URL#',
+        help="use git to download source from repository passed in url argument")
+    subparser.add_argument(
+        '-V', '--version',
+        help='override derived package version')
+    subparser.add_argument('-B', '--branch',
+                           help="specify branch(es) of git repository. "
+                                "Separate multiple branches with space. "
```

Comment:
Multiple `-B` (or `--branch`) options should be allowed in the same way we support `tags`.

An example where the space-separated approach doesn't work as expected:

```
$ spack create -n mpileaks-git -g https://github.com/LLNL/mpileaks.git -B master https://github.com/LLNL/mpileaks/releases/download/v1.0/mpileaks-1.0.tar.gz
```